homeassistant/components/screenlogic/services.py

```diff
@@ -80,8 +144,10 @@ def async_unload_screenlogic_services(hass: HomeAssistant):
         # There is still another config entry for this domain, don't remove services.
         return
 
-    if not hass.services.has_service(DOMAIN, SERVICE_SET_COLOR_MODE):
+    if not hass.services.async_services().get(DOMAIN):
```

Comment:
`async_services()` enumerates and makes a copy of all the services it would be would be better to do an `any` expression with `has_service` or check each one in a loop and remove

Reply:
Taking this a step further, would it be preferable to not even register services that aren't applicable to the hardware configuration? ie. If a salt chlorine generator is not configured, don't register the super chlorination services.

The pool controller does nothing with a command that isn't applicable so as it stands now, all services are registered regardless. But if we're looping through and running checks for each service, I could expand that pattern to registration as well.

Reply:
If we don't have any pools that support the service, its better not to register it

Reply:
I also have https://github.com/home-assistant/core/pull/108671 in flight to reduce the overhead for fetching services by domain

Reply:
`hass.services.async_services_for_domain` is now available